Src/coffeescript/cql-execution/src/elm/quantity.coffee

```diff
@@ -19,24 +21,89 @@ module.exports.Quantity = class Quantity extends Expression
+    if other instanceof Quantity and other.unit == @unit
+      @value <= parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration <= otherSmallestDuration
+    else null
 
   sameOrAfter: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value >= parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value >= parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration >= otherSmallestDuration
+    else null
 
   after: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value > parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value > parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration > otherSmallestDuration
+    else null
 
   before: (other) ->
-    if other instanceof Quantity and other.unit == @unit then @value < parseFloat other.value else null
+    if other instanceof Quantity and other.unit == @unit
+      @value < parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration < otherSmallestDuration
+    else null
     
   equals: (other) ->
-    if other.unit? && other.value? then @unit == other.unit && @value == parseFloat other.value else null
-
-time_units = {'years':'year',  'months': 'month',  'days' :'day', 'minutes': 'minute', 'seconds':'seconds', 'milliseconds' : 'millisecond' }
+    if other instanceof Quantity and @unit == other.unit
+      @value == parseFloat other.value
+    else if other instanceof Quantity and time_units[other.unit]? and time_units[@unit]?
+      thisSmallestDuration = smallestDuration(@)
+      otherSmallestDuration = smallestDuration(other)
+      thisSmallestDuration == otherSmallestDuration
+    else null
+
+# Hash of time units and their UCUM equivalents, both case-sensitive and case-insensitive
+# See http://unitsofmeasure.org/ucum.html#para-31
+# The CQL specification says that dates are based on the Gregorian calendar
+time_units = {'years': 'a_g', 'year': 'a_g', 'YEARS': 'a_g', 'YEAR': 'a_g'
+  , 'a': 'a_j', 'ANN': 'a_j', 'ann': 'a_j', 'A': 'a_j'
+  , 'months': 'mo_g', 'month':'mo_g'
+  , 'mo': 'mo_j', 'MO': 'mo_j'
+  , 'weeks': 'week', 'week': 'week', 'wk', 'week', 'WK', 'week'
```

Comment:
Another good catch.  When it was converted to JS the colons were added correctly.